src/Sylius/Bundle/ShopBundle/Grid/ProductGrid.php

```diff
@@ -0,0 +1,99 @@
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ShopBundle\Grid;
+
+use Sylius\Bundle\GridBundle\Builder\Field\DateTimeField;
+use Sylius\Bundle\GridBundle\Builder\Field\Field;
+use Sylius\Bundle\GridBundle\Builder\Field\StringField;
+use Sylius\Bundle\GridBundle\Builder\Filter\StringFilter;
+use Sylius\Bundle\GridBundle\Builder\GridBuilderInterface;
+use Sylius\Bundle\GridBundle\Grid\AbstractGrid;
+use Sylius\Bundle\GridBundle\Grid\ResourceAwareGridInterface;
+use Sylius\Component\Channel\Context\ChannelContextInterface;
+use Sylius\Component\Locale\Context\LocaleContextInterface;
+use Sylius\Component\Taxonomy\Repository\TaxonRepositoryInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Webmozart\Assert\Assert;
+
+final class ProductGrid extends AbstractGrid implements ResourceAwareGridInterface
+{
+    private ?Request $resquest;
+
+    public function __construct(
+        private string $resourceClass,
+        private ChannelContextInterface $channelContext,
+        private TaxonRepositoryInterface $taxonRepository,
+        private LocaleContextInterface $localeContext,
+        RequestStack $requestStatck,
+        private bool  $includeAllDescendants,
+    ) {
+        $this->resquest = $requestStatck->getMainRequest();
+    }
+
+    public static function getName(): string
+    {
+        return 'sylius_shop_product';
+    }
+
+    public function buildGrid(GridBuilderInterface $gridBuilder): void
+    {
+        Assert::notNull($this->resquest);
+        $localeCode = $this->localeContext->getLocaleCode();
+
+        $gridBuilder
+            ->setRepositoryMethod('createShopListQueryBuilder', [
+                $this->channelContext->getChannel(),
+                $this->taxonRepository->findOneBySlug(
+                    $this->resquest->attributes->get('slug'),
+                    $localeCode,
+                ),
+                $localeCode,
+                $this->resquest->query->get('sorting') ?? [],
+                $this->includeAllDescendants,
+            ])
+            ->orderBy('position', 'asc')
+            ->setLimits([
+                9,
+                18,
+                27,
+            ])
+            ->addField(
+                DateTimeField::create('createdAt')
+                    ->setSortable(true),
+            )
+            ->addField(
+                StringField::create('position')
+                    ->setSortable(true, 'productTaxon.position'),
+            )
+            ->addField(
+                StringField::create('name')
+                    ->setSortable(true, 'translation.name'),
+            )
+            ->addField(
+                Field::create('price', 'int')
+                    ->setSortable(true, 'channelPricing.price'),
+            )
+            ->addFilter(
+                StringFilter::create('search', ['translation.name'])
```

Comment:
In the yaml file there is a type `'shop_string'` is that the same?

Reply:
From what I can see it is the same, the only thing I can think of is that it has been split to allow a separate template, but that's just a guess.